### PR TITLE
[NEW] Show department field on Livechat visitor panel

### LINKED ISSUE #13371: Livechat Department in Room Info
In the information screen of the chat room does not appear which department the guest chose when they entered the chat.

### PR DESCRIPTION
**Pull request by renatobecker**
Closes #13371

This PR adds the `department` field of the current Livechat room on Visitor info painel, like the image below:
screen shot 2019-02-20 at 15 17 42

### CODE REVIEW

File: packages/rocketchat-livechat/client/views/app/tabbar/visitorInfo.js [hunk 1]
```diff
@@ -243,18 +250,17 @@ Template.visitorInfo.onCreated(function() {
...
-			} else {
-				this.visitorId.set();
-			}
+			const room = Rooms.findOne(currentData.rid);
+			this.visitorId.set(room && room.v && room.v._id);
+			this.departmentId.set(room && room.departmentId);
 		});
 
 		this.subscribe('livechat:visitorInfo', { rid: currentData.rid });
+		this.subscribe('livechat:departments');
```
Comment (sampaiodiego) [inline packages/rocketchat-livechat/client/views/app/tabbar/visitorInfo.js]: you don't need to subscribe for ALL departments and filter only on client. you can pass the department id to `livechat:departments` and subscribe only the department you need

File: packages/rocketchat-livechat/client/views/app/tabbar/visitorInfo.js [hunk 2]
```diff
@@ -243,18 +250,17 @@ Template.visitorInfo.onCreated(function() {
...
 	}
 
 	this.autorun(() => {
 		this.user.set(LivechatVisitor.findOne({ _id: this.visitorId.get() }));
+		this.department.set(LivechatDepartment.findOne({ _id: this.departmentId.get() }));
```
Comment (sampaiodiego) [inline packages/rocketchat-livechat/client/views/app/tabbar/visitorInfo.js]: if you'll use the department only in one helper, it is better to just return the find on the helper itself. having a reactive var like this is better if you need to use the value in multiples places, so you don't need to do multiple finds which saves in performance.

File: packages/rocketchat-livechat/client/views/app/tabbar/visitorInfo.js (hunk 1 above)
Comment (renatobecker) [reply packages/rocketchat-livechat/client/views/app/tabbar/visitorInfo.js]: Fixed.

File: packages/rocketchat-livechat/client/views/app/tabbar/visitorInfo.js (hunk 2 above)
Comment (renatobecker) [reply packages/rocketchat-livechat/client/views/app/tabbar/visitorInfo.js]: Fixed.

**Comment by juliomac** [comment]
Hi @rodrigok ,

How can I get the department information on REST API or, better, on Real-time API?

**Comment by renatobecker** [comment]
@juliomac, please take a look at our Livechat Docs:

https://rocket.chat/docs/developer-guides/rest-api/livechat/

**Comment by juliomac** [comment]
Dear @renatobecker, thanks for so quick response!

Before finding this merge and you, I did a bit of homework and went looking through all options in rest-api/livechat. There's call for listing departments, there's calls for retrieving department info, but I could not find any call retrieving what is the exact department this particular visitor came from. 

Can you give me a light? What is the call that retrieves the department defined by the below function on the target website? 

        RocketChat(function() {
         this.setDepartment('FILL HERE DEPARTMENT NAME - case sensitive');
       });

**Comment by juliomac** [comment]
Examples of calls that didn't work for me:
curl http://localhost:4000/api/v1/livechat/room?token=< tokenFromVisitor>
   >>> No department information

curl http://localhost:4000/api/v1/livechat/visitor/<tokenFromVisitor>
     >>> No Room nor department information

**Comment by renatobecker** [comment]
@juliomac, I suggest you create a new issue describing your use case, what do you think?
Thanks.

**Comment by juliomac** [comment]
Thanks, done it through: https://github.com/RocketChat/Rocket.Chat/issues/13766

**Pull request merged**